lti_consumer/lti_consumer.py

```diff
@@ -0,0 +1,829 @@
+        default=False,
+        scope=Scope.settings
+    )
+    weight = Float(
+        display_name="Weight",
+        help=(
+            "Enter the number of points possible for this component.  "
+            "The default value is 1.0.  "
+            "This setting is only used when Scored is set to True."
+        ),
+        default=1.0,
+        scope=Scope.settings,
+        values={"min": 0},
+    )
+    module_score = Float(
+        help="The score kept in the xblock KVS -- duplicate of the published score in django DB",
+        default=None,
+        scope=Scope.user_state
+    )
+    score_comment = String(
+        help="Comment as returned from grader, LTI2.0 spec",
+        default="",
+        scope=Scope.user_state
+    )
+    hide_launch = Boolean(
+        display_name="Hide External Tool",
+        help=(
+            "Select True if you want to use this component as a placeholder for syncing with an external grading  "
+            "system rather than launch an external tool.  "
+            "This setting hides the Launch button and any IFrames for this component."
+        ),
+        default=False,
+        scope=Scope.settings
+    )
+    accept_grades_past_due = Boolean(
+        display_name="Accept grades past deadline",
+        help="Select True to allow third party systems to post grades past the deadline.",
+        default=True,
+        scope=Scope.settings
+    )
+    # Users will be presented with a message indicating that their e-mail/username would be sent to a third
+    # party application. When "Open in New Page" is not selected, the tool automatically appears without any
+    # user action.
+    ask_to_send_username = Boolean(
+        display_name="Request user's username",
+        # Translators: This is used to request the user's username for a third party service.
+        help="Select True to request the user's username.",
+        default=False,
+        scope=Scope.settings
+    )
+    ask_to_send_email = Boolean(
+        display_name="Request user's email",
+        # Translators: This is used to request the user's email for a third party service.
+        help="Select True to request the user's email address.",
+        default=False,
+        scope=Scope.settings
+    )
+
+    # StudioEditableXBlockMixin configuration of fields editable in Studio
+    editable_fields = (
+        'display_name', 'description', 'lti_id', 'launch_url', 'custom_parameters', 'launch_target', 'button_text',
+        'modal_height', 'modal_width', 'has_score', 'weight', 'hide_launch', 'accept_grades_past_due',
+        'ask_to_send_username', 'ask_to_send_email'
+    )
+
+    @property
+    def descriptor(self):
+        """
+        Returns this XBlock object.
+
+        This is for backwards compatibility with the XModule API.
+        Some LMS code still assumes a descriptor attribute on the XBlock object.
+        See courseware.module_render.rebind_noauth_module_to_user.
```

Comment:
I don't think we ever had to add this to an XBlock.  If this is only required in `rebind_noauth_module_to_user()`, it might make sense to fix that function instead.  If there are more places where this is still needed, we could add the function to the `XBlock` base class, so we only have to remove it in a single place after edx-platform is fixed.